node/src/components/rpc_server.rs

```diff
@@ -219,14 +230,16 @@ where
         event: Self::Event,
     ) -> Effects<Self::Event> {
         match event {
```

Comment:
Couldn't we put that `self.enable` here? before we even match on the event?

Reply:
Ah, b/c we have requests/events for different server types at the same "level" of types. We would have to differentiate between them somehow. We already kind of do - `Event::RpcRequest(RpcRequest)`. If we had different wrapper types/enum variants for other server types we wouldn't have to write all of these `if self.enable` for every individual variant.

Reply:
I think I would go for the second option.